src/gui/qgsmapcanvas.h

```diff
@@ -72,7 +73,7 @@ class QgsReferencedRectangle;
  * Map canvas is a class for displaying all GIS data types on a canvas.
  */
 
-class GUI_EXPORT QgsMapCanvas : public QGraphicsView
+class GUI_EXPORT QgsMapCanvas : public QGraphicsView, public QgsAbstractTemporal
```

Comment:
I suspect this move will make things difficult later -- if you use inheritance here, it will be clumsy to respond to changes in the time range. (you'd need to make the setters virtual and override them here). Composition may be a better move for the map canvas class.

Reply:
... and if there is a place where it's being considered to make a setter virtual it's almost always better to not make it virtual and call an internal virtual (private) method instead. Just sayin'.